src/DataTypes/DataTypeDateTime.cpp

```diff
@@ -20,15 +20,15 @@
 namespace
 {
 using namespace DB;
-inline void readText(time_t & x, ReadBuffer & istr, const FormatSettings & settings, const DateLUTImpl & time_zone, const DateLUTImpl & utc_time_zone)
+inline void readText(time_t & x, ReadBuffer & istr, const FormatSettings & settings, const TimeZoneImpl & time_zone, const TimeZoneImpl & utc_time_zone)
 {
     switch (settings.date_time_input_format)
     {
         case FormatSettings::DateTimeInputFormat::Basic:
-            readDateTimeText(x, istr, time_zone);
+            readDateTimeText(x, istr, time_zone.getDefaultLUT());
```

Comment:
It's too complex to know about both DateLUTImpl and TimeZoneImpl in code. I think we should change readDateTimeText() and others interface too. We need hide DateLUTImpl with our new class at all. It's too error prone to use both of them.